### DISCUSSION TITLE
Creating a dynamic redirect in nuxtJS

### DISCUSSION THREAD

**Discussion by ahoiroman** [nuxt/nuxt#24455] [Questions] [answered]
I'd like to solve the following problem:

A user should click a link and should be redirected to a target url. But: the link should not be visible, but resolved by querying the API, getting back the target URL and redirect than.

## Example

Link: https://my-app.test/redirect/123123123

redirects to the url, which is saved for an item with the id 123123123.

## What I did so far

I created a middleware called `redirect.ts` with the following content:

```
export default defineNuxtRouteMiddleware(async (to, from) => {
    const {view} = useItem()

    if (!to.params.id) {
        return abortNavigation();
    }

    const {data, error} = await view(to.params.id)

    if (error.value || !data.value?.url) {
        abortNavigation();
    }


    navigateTo(data.value?.url, {external: true})
});
```

Now I created a file like this:

`pages/item/[id]/redirect.vue`

```
<script setup lang="ts">
definePageMeta({
  middleware: 'redirect',
});
</script>
```

And this works as expected. At least, almost like expected.

## The problem

The user clicks the link, a blank page is being shown on redirect. If the user uses the "back" button of the browser, the blank page is still shown.

## What I want to see

The redirect should be executed without an extra page.

**Comment by masyoudi** [comment, marked as answer]
maybe you need to create server routes


`/server/routes/item/[id]/redirect.get.ts`
```ts
export default defineEventHandler(async (event) => {
  const {url} = await serverFetch(`/to/api/${event.context.params?.id}`);
  return sendRedirect(event, url);
});
```

and the link should use tag `a`
```html
<a href="/item/123/redirect">Redirect</a>
```

**Reply by ahoiroman** [reply]
This seems to be a good idea, thank you.

However, as a downside, I can not use my useOffer composable here and I _need_ to use a NuxtLink, as I am using nuxt/ui-pro 's  LandingCard component (https://ui.nuxt.com/pro/components/landing/landing-card), which basically places an NuxtLink with absolute position and inset-0 over the card.

However, it binds NuxtLinkProps like `external`. If I use `external` like this:

```
  <ULandingCard external :to="`/item/${item.id}/redirect`">
```

it _works_. But it shows 

> [Vue Router warn]: No match found for location with path "/item/123/redirect"

in the console.

**Reply by masyoudi** [reply]
you  can use these props `href="/item/123/redirect" target="_parent" no-prefetch`

**Reply by masyoudi** [reply]
```html
<ULandingCard :href="`/item/${item.id}/redirect`" target="_top" no-prefetch>
```
